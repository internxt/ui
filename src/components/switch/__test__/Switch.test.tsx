import React from 'react';
import { describe, expect, it, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SwitchComponent } from '../';

describe('Switch component', () => {
  it('Switch onClick should be called correctly', () => {
    const switchClick = vi.fn();
    render(<SwitchComponent size="md" onClick={switchClick} />);
    const switchComponent = screen.getByTestId('switch');
    fireEvent.click(switchComponent);
    expect(switchClick).toHaveBeenCalledOnce();
  });

  it('Switch onCheckedChange should be called with the correct value', async () => {
    const onCheckedChangeMock = vi.fn();
    render(<SwitchComponent size="md" onCheckedChange={onCheckedChangeMock} />);

    const switchComponent = screen.getByTestId('switch');

    await userEvent.click(switchComponent);

    expect(onCheckedChangeMock).toHaveBeenCalledWith(true);

    await userEvent.click(switchComponent);

    expect(onCheckedChangeMock).toHaveBeenCalledWith(false);
  });

  it('Medium switch should render correctly by default size', () => {
    const switchComponent = render(<SwitchComponent />);
    expect(switchComponent).toMatchSnapshot();
  });

  it('Medium switch should render correctly', () => {
    const switchComponent = render(<SwitchComponent size="md" />);
    expect(switchComponent).toMatchSnapshot();
  });

  it('Medium switch disabled should render correctly', () => {
    const switchComponent = render(<SwitchComponent disabled size="md" />);
    expect(switchComponent).toMatchSnapshot();
  });

  it('Large switch should render correctly', () => {
    const switchComponent = render(<SwitchComponent size="lg" />);
    expect(switchComponent).toMatchSnapshot();
  });

  it('Large switch disabled should render correctly', () => {
    const switchComponent = render(<SwitchComponent disabled size="lg" />);
    expect(switchComponent).toMatchSnapshot();
  });

  it('Extra Large switch should render correctly', () => {
    const switchComponent = render(<SwitchComponent size="xl" />);
    expect(switchComponent).toMatchSnapshot();
  });

  it('Extra Large switch disabled should render correctly', () => {
    const switchComponent = render(<SwitchComponent disabled size="xl" />);
    expect(switchComponent).toMatchSnapshot();
  });
});
