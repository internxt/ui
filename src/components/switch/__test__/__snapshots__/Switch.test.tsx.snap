// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Switch component > Extra Large switch disabled should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="relative inline-flex items-center cursor-pointer w-14 h-8"
        data-testid="switch"
      >
        <input
          class="sr-only"
          disabled=""
          type="checkbox"
        />
        <div
          class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-5 data-[state=checked]:bg-green/50"
        />
        <span
          class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-7 h-7 pointer-events-none"
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="relative inline-flex items-center cursor-pointer w-14 h-8"
      data-testid="switch"
    >
      <input
        class="sr-only"
        disabled=""
        type="checkbox"
      />
      <div
        class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-5 data-[state=checked]:bg-green/50"
      />
      <span
        class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-7 h-7 pointer-events-none"
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch component > Extra Large switch should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="relative inline-flex items-center cursor-pointer w-14 h-8"
        data-testid="switch"
      >
        <input
          class="sr-only"
          type="checkbox"
        />
        <div
          class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-10 "
        />
        <span
          class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-7 h-7 "
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="relative inline-flex items-center cursor-pointer w-14 h-8"
      data-testid="switch"
    >
      <input
        class="sr-only"
        type="checkbox"
      />
      <div
        class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-10 "
      />
      <span
        class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-7 h-7 "
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch component > Large switch disabled should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="relative inline-flex items-center cursor-pointer w-12 h-7"
        data-testid="switch"
      >
        <input
          class="sr-only"
          disabled=""
          type="checkbox"
        />
        <div
          class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-5 data-[state=checked]:bg-green/50"
        />
        <span
          class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-6 h-6 pointer-events-none"
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="relative inline-flex items-center cursor-pointer w-12 h-7"
      data-testid="switch"
    >
      <input
        class="sr-only"
        disabled=""
        type="checkbox"
      />
      <div
        class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-5 data-[state=checked]:bg-green/50"
      />
      <span
        class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-6 h-6 pointer-events-none"
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch component > Large switch should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="relative inline-flex items-center cursor-pointer w-12 h-7"
        data-testid="switch"
      >
        <input
          class="sr-only"
          type="checkbox"
        />
        <div
          class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-10 "
        />
        <span
          class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-6 h-6 "
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="relative inline-flex items-center cursor-pointer w-12 h-7"
      data-testid="switch"
    >
      <input
        class="sr-only"
        type="checkbox"
      />
      <div
        class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-10 "
      />
      <span
        class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-6 h-6 "
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch component > Medium switch disabled should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="relative inline-flex items-center cursor-pointer w-8 h-5"
        data-testid="switch"
      >
        <input
          class="sr-only"
          disabled=""
          type="checkbox"
        />
        <div
          class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-5 data-[state=checked]:bg-green/50"
        />
        <span
          class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-4 h-4 pointer-events-none"
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="relative inline-flex items-center cursor-pointer w-8 h-5"
      data-testid="switch"
    >
      <input
        class="sr-only"
        disabled=""
        type="checkbox"
      />
      <div
        class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-5 data-[state=checked]:bg-green/50"
      />
      <span
        class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-4 h-4 pointer-events-none"
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch component > Medium switch should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="relative inline-flex items-center cursor-pointer w-8 h-5"
        data-testid="switch"
      >
        <input
          class="sr-only"
          type="checkbox"
        />
        <div
          class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-10 "
        />
        <span
          class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-4 h-4 "
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="relative inline-flex items-center cursor-pointer w-8 h-5"
      data-testid="switch"
    >
      <input
        class="sr-only"
        type="checkbox"
      />
      <div
        class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-10 "
      />
      <span
        class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-4 h-4 "
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch component > Medium switch should render correctly by default size 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="relative inline-flex items-center cursor-pointer w-8 h-5"
        data-testid="switch"
      >
        <input
          class="sr-only"
          type="checkbox"
        />
        <div
          class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-10 "
        />
        <span
          class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-4 h-4 "
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="relative inline-flex items-center cursor-pointer w-8 h-5"
      data-testid="switch"
    >
      <input
        class="sr-only"
        type="checkbox"
      />
      <div
        class="absolute inset-0 rounded-full transition-colors duration-200 bg-gray-10 "
      />
      <span
        class="block bg-white rounded-full transition-transform duration-200 transform translate-x-[3px] w-4 h-4 "
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
